projects/wallet-template/src/containers/WalletPopup/pages/Accounts/ImportAccounts.tsx

```diff
@@ -0,0 +1,349 @@
+    mode: "onSubmit",
+    reValidateMode: "onSubmit",
+    defaultValues: {
+      scheme: "Sr25519",
+      networks: {
+        polkadot: false,
+        westend: false,
+        kusama: false,
+      },
+    },
+  })
+
+  const handleChange = (chain: "kusama" | "polkadot" | "westend") => {
+    const newValue = !getValues(`networks.${chain}`)
+    return {
+      ...getValues("networks"),
+      [chain]: newValue,
+    }
+  }
+
+  const onActiveTabChanged = (tab: Tab) => {
+    setActiveTab(tab)
+    clearErrors("key")
+    setValue("key", "")
+  }
+
+  const onSubmit: SubmitHandler<FormFields> = async (data) => {
+    switch (activeTab) {
+      case "private": {
+        await rpc.client.insertKeyset({
+          _type: "PrivateKey",
+          name: data.keysetName,
+          scheme: data.scheme,
+          miniSecret: bytesToHex(entropyToMiniSecret(fromHex(data.key!))),
+          privatekey: data.key!,
+          createdAt: Date.now(),
+        })
+        break
+      }
+      case "mnemonic": {
+        const entropy = mnemonicToEntropy(data.mnemonicInput!)
+        const miniSecret = entropyToMiniSecret(entropy)
+        // TODO: Add support for Ed25519 and Ecdsa
+        const derive = sr25519CreateDerive(miniSecret)
+        const derivationPaths = Object.entries(data.networks)
+          .filter(([_, selected]) => selected)
+          .map(([networkId, _]) => {
+            const network = networks.find(
+              ({ label }) => networkId.toLowerCase() === label.toLowerCase(),
+            )!
+            const path = `//${network.value}//0`
+            return {
+              chainId: network.chainId,
+              path,
+              publicKey: toHex(derive(`//${network.value}//0`).publicKey),
+            }
+          })
+
+        await rpc.client.insertKeyset({
+          _type: "DerivationPath",
+          name: data.keysetName,
+          scheme: data.scheme,
+          miniSecret: bytesToHex(miniSecret),
+          derivationPaths,
+          createdAt: Date.now(),
+        })
+        break
+      }
+    }
+    window.localStorage.setItem("selectedKeysetName", data.keysetName)
+    navigate("/accounts")
+  }
+
+  const validatePrivateKey = (value: string | undefined) => {
+    if (!value) return "Private Key is required"
+
+    const { scheme } = getValues()
+    const bytes = fromHex(value)
+    try {
+      switch (scheme) {
+        case "Ecdsa":
+          ecdsa.getPublicKey(bytes)
+          break
+        case "Ed25519":
+          ed25519.getPublicKey(bytes)
+          break
+        case "Sr25519":
+          sr25519.getPublicKey(bytes)
+          break
+      }
+      return true
+    } catch (_) {
+      return "Invalid private format"
+    }
+  }
+
+  const validateMnemonic = (value: string | undefined) => {
+    if (!value) return "Mnemonic is required"
+
+    try {
+      mnemonicToEntropy(value)
+      return true
+    } catch (_) {
+      return "Invalid mnemonic"
+    }
+  }
```

Comment:
`validateMnemonic` should be outside this component fn so it's not re-created on every render